src/CSModule.sol

```diff
@@ -999,8 +1032,19 @@ contract CSModule is
     }
 
     /// @inheritdoc ICSModule
-    function depositQueueItem(uint128 index) external view returns (Batch) {
-        return depositQueue.at(index);
+    function depositQueuePointers(
+        uint256 queuePriority
+    ) external view returns (uint128 head, uint128 tail) {
+        QueueLib.Queue storage q = _getQueue(queuePriority);
+        return (q.head, q.tail);
```

Comment:
What is the point of the method? Maybe it is easier to make `_queueByPriority` public?

Reply:
Or make `legacyQueue` internal and make `_getQueue()` public

Reply:
`_getQueue` has to return a `QueueLib.Queue` via storage to make the queue mapping accessible; but s*lidity doesn't allow for public functions to return variable with storage location. At the same time we can't make `_queueByPriority` public because of the legacy queue, that's why we have `_getQueue` in the first place.

So the function acts as `depositQueue` method implemented in the previous version, but it clearly states it returns only the pointers, not the queue itself (the same behaviour as before).

This method is particularly useful for off-chain tooling to inspect queue(s), and widely used in tests.